music21/stream/base.py

```diff
@@ -6858,11 +6880,53 @@ def makeAccidentals(
                     if n.tie is not None and n.tie.type != 'stop':
                         seenPitchNames.add(p.nameWithOctave)
 
+                    # handle this note-in-chord's ornaments' otherPitches
```

Comment:
this is repeated three times in a routine that is already way too long.  Can't it be handled by a separate helper function?

Reply:
btw -- `if n.pitches: p = n.pitches[-1]` will work for notes/chords/rests/unpitched etc.